### PR TITLE
If we have passed the private key, don't copy it implicitly

### LINKED ISSUE #16197: Password request for an unprotected key
The following commands
```
openssl genrsa -out key.pem 2048
openssl req -new -key key.pem -out test.csr
```
request a password for the unprotected key

It's a regression comparing OpenSSL 1.1.1

### PR DESCRIPTION
**Pull request by beldmit**
Fixes #16197

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: doc/man1/openssl-req.pod.in [hunk 1]
```diff
@@ -205,7 +205,8 @@ See L<openssl-format-options(1)> for details.
 =item B<-keyout> I<filename>
 
 This gives the filename to write any private key to that has been newly created
-or read from B<-key>.
+or read from B<-key>. If you want to write the private key and the B<-key> option
+is provided, you should provide the B<-keyout> option explicitly.
```
Comment (DDvO) [inline doc/man1/openssl-req.pod.in:209]: You also need to update the sentence thereafter, which is:
```
If the B<-keyout> option is not given the filename specified in the
configuration file with the B<default_keyfile> option is used, if present.
```
Better do this before your new sentence, for instance as follows:
```
If neither the B<-keyout> option nor the B<-key> option are given the filename specified
in the configuration file with the B<default_keyfile> option is used, if present.
Thus, if you want to write the private key and the B<-key> option
is provided, you should provide the B<-keyout> option explicitly.
```

**Review by DDvO — CHANGES_REQUESTED** [review]
In the doc a further update is needed.

File: doc/man1/openssl-req.pod.in (hunk 1 above)
Comment (beldmit) [reply doc/man1/openssl-req.pod.in]: I've updated the text. @t8m, @DDvO - please reapprove.

**Review by DDvO — APPROVED** [review]
LGTM

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by beldmit** [comment]
Merged. Thanks!

**Pull request closed without merging**